app/src/main/java/sensors_in_paradise/sonar/page2/labels_editor/LabelsEditorDialog.kt

```diff
@@ -0,0 +1,295 @@
+        rangeSlider.valueTo = rangeEnd.toFloat()
+        Log.d(
+            "LabelsEditorDialog",
+            "setActivitySelected rangeStart $rangeStart and rangeEnd $rangeEnd"
+        )
+
+        rangeSlider.values = arrayListOf(
+            editableRecording.getRelativeStartTimeOfActivity(selectedItemIndex).toFloat(),
+            editableRecording.getRelativeEndTimeOfActivity(selectedItemIndex).toFloat()
+        )
+        startTV.text = GlobalValues.getDurationAsString(rangeStart)
+        endTV.text = GlobalValues.getDurationAsString(rangeEnd)
+
+        val startTime =
+            editableRecording.getRelativeStartTimeOfActivity(selectedItemIndex).toFloat()
+        val endTime = editableRecording.getRelativeEndTimeOfActivity(selectedItemIndex).toFloat()
+
+        Log.d("LabelsEditorDialog", "setActivitySelected startTime $startTime and endTime $endTime")
+        rangeSlider.values = arrayListOf(
+            startTime,
+            endTime
+        )
+    }
+    private fun loopVisualizerForSelectedActivity() {
+        visualizer.loopInterval(
+            editableRecording.relativeSensorTimeToVideoTime(
+                editableRecording.getRelativeStartTimeOfActivity(
+                    selectedItemIndex
+                )
+            ),
+            editableRecording.relativeSensorTimeToVideoTime(
+                editableRecording.getRelativeEndTimeOfActivity(
+                    selectedItemIndex
+                )
+            )
+        )
+    }
+    private fun notifyNewActivitySelected() {
+        updateRangeSlider()
+        statusTV.text =
+            if (editableRecording.areTimeStampsConsistent()) "" else "inconsistent timestamps"
+        loopVisualizerForSelectedActivity()
+    }
+
+    private fun showActivitiesDialog() {
+        if (activitiesDialog == null) {
+            activitiesDialog = PersistentCategoriesDialog(
+                context,
+                "Select an activity label",
+                GlobalValues.getActivityLabelsJSONFile(context),
+                defaultItems = GlobalValues.DEFINED_ACTIVITIES,
+                callback = { value ->
+                    activities[selectedItemIndex].activity = value
+                    carousel.refresh()
+                },
+            )
+        }
+        activitiesDialog?.show(true)
+    }
+
+    private fun onActivityInserted(index: Int) {
+        Log.d(
+            "LabelsEditorDialog",
+            "onActivityInserted at index $index while selectedItemIndex is $selectedItemIndex"
+        )
+        if (index <= selectedItemIndex) {
+            selectedItemIndex += 1
+            carousel.jumpToIndex(selectedItemIndex)
+        }
+        carousel.refresh()
+    }
+
+    private fun onActivityRemoved(index: Int) {
+        Log.d(
+            "LabelsEditorDialog",
+            "onActivityRemoved at index $index while selectedItemIndex is $selectedItemIndex"
+        )
+        assert(index != selectedItemIndex)
```

Comment:
lieber try catch als assert in der endversion?

Reply:
jut